### PR TITLE
Add support for User Messaging Platform SDK on Android

### PR DESCRIPTION
**Pull request by jonsimantov**
Android implementation of UMP, mostly via a ConsentInfoHelper class.

Also adds OperationInProgress errors to iOS SDK.

***

Integration tests running in this PR, plus manual testing on Android emulator.

***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [X] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

**Comment by jonsimantov** [comment]
PTAL for code changes - although there is a build hang issue on Windows that I need to figure out before merging this.

File: gma/src_java/com/google/firebase/gma/internal/cpp/ConsentInfoHelper.java [hunk 1]
```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright 2021 Google LLC
```
Comment (a-maurice) [inline gma/src_java/com/google/firebase/gma/internal/cpp/ConsentInfoHelper.java:2]: 3

File: messaging/messaging_java/build.gradle [hunk 2]
```diff
@@ -81,6 +81,14 @@ afterEvaluate {
              'https://github.com/google/flatbuffers.git',
              flatbuffersDir
       }
+      exec {
```
Comment (a-maurice) [inline messaging/messaging_java/build.gradle:84]: This should probably be in a separate PR

**Review by a-maurice — APPROVED** [review]

File: gma/src/android/ump/consent_info_internal_android.cc [hunk 3]
```diff
@@ -0,0 +1,667 @@
...
+      ENUM_VALUE(consent_info_helper, FunctionLoadAndShowConsentFormIfRequired);
+  enums_.function_show_privacy_options_form =
+      ENUM_VALUE(consent_info_helper, FunctionShowPrivacyOptionsForm);
+  enums_.function_count = ENUM_VALUE(consent_info_helper, FunctionCount);
+}
+
+void ConsentInfoInternalAndroid::JNI_ConsentInfoHelper_completeFuture(
+    JNIEnv* env, jclass clazz, jint future_fn, jlong consent_info_internal_ptr,
+    jlong future_handle, jint error_code, jobject error_message_obj) {
+  MutexLock lock(s_instance_mutex);
```
Comment (a-maurice) [inline gma/src/android/ump/consent_info_internal_android.cc:229]: I think there might be a problem with the locks.  If the destructor is called, it gets the lock on s_instance_mutex, and then calls disconnect, which locks the java object.  In some of the java calls, it locks the java object, and then calls completeFuture, which locks s_instance_mutex.

File: messaging/messaging_java/build.gradle (hunk 2 above)
Comment (jonsimantov) [reply messaging/messaging_java/build.gradle:84]: Added [here](https://github.com/firebase/firebase-cpp-sdk/pull/1444) - I'll keep this in this PR (so my build works) until that PR goes in.

File: gma/src/android/ump/consent_info_internal_android.cc (hunk 3 above)
Comment (jonsimantov) [reply gma/src/android/ump/consent_info_internal_android.cc:231]: Yeah, sounds like this could cause a deadlock, good catch. I think if I do the Disconnect first, prior to locking s_instance, it should be safe.
